File: 75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int i=0;//0 pointer
        int j= nums.length-1; // pointer for 2
        int k =0; // current pointer
        while(k<=j)
        {
            if(nums[k]==0)
            {
                swap(nums,k,i);
                i++;
                k++;
            }
            else if(nums[k]==2)
            {
                swap(nums,k,j);
                j--;
            }
            else{
                k++;
            }
        }
    }
    void swap(int[] nums,int i,int j)
    {
      int k = nums[i];
      nums[i] =nums[j];
      nums[j]=k;
    }
}

File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int low = 0;
        int high = nums.size()-1;
        int i =0;
        while(i<=high && low<=high)
        {
            if(nums[i]==0){
                swap(nums,i,low);
                low++;
                i++;
            }
            else if(nums[i]==2)
            {
                swap(nums,i,high);
                high--;
            }
            else i++;
        }
        return;
    }
    void swap(vector<int>& ans, int i, int j)
    {
        int temp = ans[i];
        ans[i] = ans[j];
        ans[j] =temp;
    }

};